src/main/java/scala/compat/java8/ScalaStreaming.java

```diff
@@ -0,0 +1,905 @@
+package scala.compat.java8;
+
+import scala.compat.java8.converterImpl.*;
+import scala.compat.java8.collectionImpl.*;
+import java.util.stream.*;
+import scala.compat.java8.runtime.CollectionInternals;
+
+public class ScalaStreaming {
```

Comment:
This could use some comments here or on this individual methods about immutability, or otherwise the binding policy of the generated streams.